### ISSUE TITLE
How to add contentBody in OpenAPI json when using starlette.Request directly to stream binary content

### ISSUE THREAD

**Issue by chris-allnutt** [tiangolo/fastapi#1135]
### First check

* [x ] I used the GitHub search to find a similar issue and didn't find it.
* [ x] I searched the FastAPI documentation, with the integrated search.
* [ x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

I've got a route setup that uses the starlette Request directly so I can stream a large binary file without multipart form data, we describe some headers and a stream which I consume, everything works, however I don't see a request body being generated in the outputted json.  The code works and I'm able to upload the file, but the api doesn't add my expected requestBody

Expected Swagger Output:
```
paths:
  /streams:
    post:
      requestBody:
        content:
          application/binary:
            schema:
                title: "Raw Binary Stream",
                type: "string",
                format: "binary"

```
The following code provides open api with json for header parameters but not the binary request body schema:
```
@app.post("/streams", response_model=Stream)
async def upload_stream(
    id=Header(...),
    description=Header(...),
    streamData: Request = Body(..., media_type="application/binary", title="Raw Binary Stream"),
):
    with open('/tmp/local_file_location.binary', "wb") as stream_file:
        async for chunk in streamData.stream():
            stream_file.write(chunk)
```

Using the following gives me the openapi I want, but now fastapi is processing the body into streamData which appears to also alter the content as I get a utf-8 error when i attempt to use the stream from request :

```
@app.post("/streams", response_model=Stream)
async def upload_stream(
    id=Header(...),
    description=Header(...),
    streamData: bytes = Body(
        ..., media_type="application/binary", title="Raw Binary Stream"
    ),
    request: Request = Body(...),
    db: asyncpg.connection.Connection = Depends(get_db),
):
    with open('/tmp/local_file_location.binary', "wb") as stream_file:
        async for chunk in request.stream():
            stream_file.write(chunk)
```

Is it possible to do either of the following?
- Use the starlette.Request directly but still properly render the requestBody schema
- Specify a requestBody Descriptor without providing a parameter to the function definition

**Comment by tiangolo** [comment]
Yeah, you can use the techniques to extend OpenAPI and add your custom schema there: https://fastapi.tiangolo.com/advanced/extending-openapi/

**Comment by chris-allnutt** [comment]
Thanks @tiangolo we'll follow that recommendation.
